static/app/components/tours/components.tsx

```diff
@@ -0,0 +1,226 @@
+import {Fragment, useEffect} from 'react';
+import {css, useTheme} from '@emotion/react';
+import styled from '@emotion/styled';
+import {FocusScope} from '@react-aria/focus';
+
+import {Button} from 'sentry/components/button';
+import {Flex} from 'sentry/components/container/flex';
+import {Overlay, PositionWrapper} from 'sentry/components/overlay';
+import {
+  type TourContextType,
+  type TourEnumType,
+  type TourState,
+  type TourStep,
+  useTourReducer,
+} from 'sentry/components/tours/tourContext';
+import {IconClose} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import useOverlay, {type UseOverlayProps} from 'sentry/utils/useOverlay';
+
+interface TourContextProviderProps<T extends TourEnumType> {
+  /**
+   * The children of the tour context provider.
+   * All children of this component will be blurred when the tour is active.
+   * Only the active tour element and overlay will be discernible.
+   */
+  children: React.ReactNode;
+  /**
+   * Whether the tour can be accessed by the user
+   */
+  isAvailable: TourState<T>['isAvailable'];
+  /**
+   * The ordered list of Step IDs
+   */
+  orderedStepIds: TourState<T>['orderedStepIds'];
+  /**
+   * The React context (from createContext) containing the provider for the tour.
+   * The value for this prop comes from useTourReducer, to avoid extra steps.
+   */
+  tourContext: React.Context<TourContextType<T>>;
+}
+
+export function TourContextProvider<T extends TourEnumType>({
+  children,
+  isAvailable,
+  tourContext: TourContext,
+  orderedStepIds,
+}: TourContextProviderProps<T>) {
+  const tourContext = useTourReducer<T>({isAvailable, orderedStepIds, currentStep: null});
+  const isTourActive = tourContext.currentStep !== null;
+  return (
+    <TourContext.Provider value={tourContext}>
+      <BlurContainer>
+        {children}
+        {isTourActive && <BlurWindow />}
```

Comment:
Is the idea that this blur window might be contained within any arbitrary element on the page? If so this is great, but if it's always going to be a full-page tour it might make more sense to use a portal to render it at the document.body level

Reply:
I imagine full page tours make the most sense, I'm not too familiar with portals but I'll take a look, thanks!

Reply:
I think this is okay for now, though portals might be the way to do it in some cases. `document.body` is probably too elevated since we don't want to blur the sidebar, but maybe the main page content? Either way, i'm okay leaving this for now since it's not too big a change either way.